program_v2/programs/sol-did/src/instructions/add_service.rs

```diff
@@ -0,0 +1,28 @@
+use crate::state::{DidAccount, Service};
+use anchor_lang::prelude::*;
+
+pub fn add_service(ctx: Context<AddService>, service: Service) -> Result<()> {
+    let data = &mut ctx.accounts.data;
+    if data.services.iter().all(|x| x.id != service.id) {
+        data.services.push(service);
+        Ok(())
+    } else {
+        Err(error!(ErrorCode::RepetitiveService))
+    }
+}
+
+#[error_code]
+pub enum ErrorCode {
+    #[msg("serviceID already exists in current service")]
+    RepetitiveService,
+}
+
+
+#[derive(Accounts)]
+pub struct AddService<'info> {
+    #[account(mut)]
+    pub data: Account<'info, DidAccount>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
```

Comment:
please remove authority and system_program from the accounts (I think they are not needed for your state of the implementation).